src/test/java/unit/WorkspaceUser.java

```diff
@@ -0,0 +1,191 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFWorkspaceUser;
+import bio.terra.workspace.model.IamRole;
+import com.fasterxml.jackson.core.type.TypeReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.SingleWorkspaceUnit;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra workspace` commands that handle sharing with other users. */
+@Tag("unit")
+public class WorkspaceUser extends SingleWorkspaceUnit {
+  @BeforeEach
+  @Override
+  protected void setupEachTime() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra workspace list-users --format=json`
+    List<UFWorkspaceUser> listWorkspaceUsers =
+        TestCommand.runCommandExpectSuccess(
+            new TypeReference<>() {}, "workspace", "list-users", "--format=json");
+
+    for (UFWorkspaceUser user : listWorkspaceUsers) {
+      if (user.email.equalsIgnoreCase(workspaceCreator.email)) {
+        continue;
+      }
+      for (IamRole role : user.roles) {
+        // `terra workspace remove-user --email=$email --role=$role
+        TestCommand.runCommandExpectSuccess(
+            "workspace", "remove-user", "--email=" + user.email, "--role=" + role);
+      }
+    }
+
+    super.setupEachTime();
```

Comment:
how do we know if we should do this last or first?

Reply:
It should be last so that the context is cleared just before each test method. I'll add a note on the superclass.